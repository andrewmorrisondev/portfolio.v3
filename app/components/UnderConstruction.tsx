// biome-ignore lint/style/useImportType: <explanation>
import React from "react";

const UnderConstruction: React.FC = () => {
  return (
    <div className="flex items-center justify-center min-h-screen bg-gray-900 text-gray-200">
      <div className="bg-gray-800 shadow-lg rounded-lg p-8 max-w-md text-center">
        <h1 className="text-3xl font-bold mb-4 text-yellow-500">
          ðŸš§ Under Construction ðŸš§
        </h1>
        <p className="text-gray-400 mb-6">
          We are working hard to bring you a great experience. Check back soon!
        </p>
      </div>
    </div>
  );
};

export default UnderConstruction;
