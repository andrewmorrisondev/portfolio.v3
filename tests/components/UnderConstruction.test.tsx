import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import UnderConstruction from '../../app/components/UnderConstruction';

describe('UnderConstruction Page', () => {
  it('renders the UnderConstruction component without crashing', () => {
    render(<UnderConstruction />);
    const heading = screen.getByRole('heading', {
      name: /under construction/i,
    });
    expect(heading).toBeInTheDocument();
  });

  it('renders the heading correctly', () => {
    render(<UnderConstruction />);
    const heading = screen.getByRole('heading', { name: /under construction/i });
    expect(heading).toHaveTextContent('ðŸš§ Under Construction ðŸš§');
  });

  it('displays the message about coming soon', () => {
    render(<UnderConstruction />);
    const message = screen.getByText(/weâ€™re working hard to bring you a great experience/i);
    expect(message).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<UnderConstruction />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an accessible heading', () => {
    render(<UnderConstruction />);
    const heading = screen.getByRole('heading');
    expect(heading).toHaveAccessibleName(/under construction/i);
  });
});
