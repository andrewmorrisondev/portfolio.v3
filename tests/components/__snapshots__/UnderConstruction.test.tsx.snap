// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnderConstruction Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center min-h-screen bg-gray-900 text-gray-200"
  >
    <div
      class="bg-gray-800 shadow-lg rounded-lg p-8 max-w-md text-center"
    >
      <h1
        class="text-3xl font-bold mb-4 text-yellow-500"
      >
        ðŸš§ Under Construction ðŸš§
      </h1>
      <p
        class="text-gray-400 mb-6"
      >
        We are working hard to bring you a great experience. Check back soon!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
